drivers/i2c/Kconfig

```diff
@@ -218,7 +220,7 @@ config I2C_1
 config I2C_1_NAME
 	string "Port 1 device name"
 	default "I2C_1"
-	depends on I2C_1
+	depends on I2C_1 && !HAS_DTS_I2C
 
 config I2C_1_DEFAULT_CFG
```

Comment:
That's also a good option. I'll open a Jira issue to give this matter more visibility.